src/main/java/com/body/controllers/PersonController.java

```diff
@@ -9,27 +9,52 @@
 import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
+import java.util.Optional;
 
 @RestController
-@RequestMapping("/users")
+@RequestMapping("/persons")
 public class PersonController {
     private final PersonService personService;
 
     public PersonController(@Autowired PersonService personService) {
         this.personService = personService;
     }
 
+    @GetMapping("/admin")
+    public List<Person> findAllPersons() {
+        return personService.findAllPersons();
+    }
+
+    @GetMapping("/admin/{id}")
+    public ResponseEntity<Person> getUserById(@PathVariable String id) {
+        Optional <Person> person = personService.getRepository().findById(id);
+        if(person.isEmpty()) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        return new ResponseEntity<>(person.get(), HttpStatus.OK);
+    }
+
     @GetMapping
-    public List<Person> getAllUsers() {
-        return personService.getAllPersons();
+    public List<Object> getAllPerson() {
+        return personService.getRepository().getAllPersons();
     }
 
-    @GetMapping("/{id}")
-    public ResponseEntity<Person> getPersonById(@PathVariable Long id) {
-        if(personService.getRepository().findById(id).isEmpty()) {
+    @GetMapping("/{firstName}")
+    public ResponseEntity<Object> getPersonByFirstName(@PathVariable String firstName) {
+        Optional <List<Object>> person = Optional.ofNullable(personService.getPersonByFirstName(firstName));
```

Comment:
Тоже самое и про firstName и про 
```suggestion
        Optional<List<Object>> person = Optional.ofNullable(personService.getPersonByFirstName(firstName));
```
Тот длинный пост если что про многое, сам глянь что и где.
Так же мне не нравится List<Object> и Object в очень многих местах, почитай про то, как это исправить. Если не справишься - пиши. (Я к тому, что это должен быть Person и пр.)